modules/pafRtdProvider.js

```diff
@@ -13,40 +13,60 @@ const SUBMODULE_NAME = 'paf';
  * @param {Object} rtdConfig
  * @param {Object} userConsent
  */
-function getBidRequestData(reqBidsConfigObj, callback, rtdConfig, userConsent) {
-
+function getBidRequestData(reqBidsConfigObj, onDone, rtdConfig, userConsent) {
+  logMessage('DEBUG(paf):', rtdConfig);
   let idsAndPreferences;
-  let seed;
   const adUnits = (reqBidsConfigObj.adUnits || getGlobal().adUnits);
 
-  if (window.PAF) {
+  if (rtdConfig.params && rtdConfig.params.proxyHostName && window.PAF) {
     idsAndPreferences = window.PAF.getIdsAndPreferences();
     if (!idsAndPreferences) {
-      callback();
+      onDone();
       return;
     }
+
+    let transaction_ids = [];
+    for (var i=0; i < adUnits.length; i++) {
+      const uuid = generateUUID();
+      transaction_ids.push(uuid)
+      deepSetValue(adUnits[i], `ortb2Imp.ext.data.paf.transaction_id`, uuid)
+    }
+
     logMessage('DEBUG(idsAndPreferences):', idsAndPreferences);
-    seed = rtdConfig.multipleTransactions ? window.PAF.getSeed() : window.PAF.getSeed(adUnits.length);
-    logMessage('DEBUG(seed):', seed);
+    window.PAF.createSeed({proxyHostName: rtdConfig.params.proxyHostName, callback: function (seed) {setData(seed, rtdConfig, onDone);}}, transaction_ids)
   } else {
-    callback();
+    onDone();
     return;
   }
+}
 
-  if (seed.transaction_ids) {
-    for (var i=0; i < adUnits.length; i++) {
-      deepSetValue(adUnits[i], `ortb2Imp.ext.data.paf_transaction_id`, seed.transaction_ids[i])
-    }
+/**
+ * Lazy merge objects.
+ * @param {Object} target
+ * @param {Object} source
+ */
+ function mergeLazy(target, source) {
+  if (!isPlainObject(target)) {
+    target = {};
   }
 
+  if (!isPlainObject(source)) {
+    source = {};
+  }
+
+  return mergeDeep(target, source);
+}
+
+function setData(seed, rtdConfig, onDone) {
```

Comment:
I don't catch yet the exactly what is going on in the positive branch of `if (rtdConfig.params && rtdConfig.params.bidders)`. However, I don't see a direct association between one Transmission-Id and an AdUnit here and I think that it should be in this branch of code, isn't it? We need a transaction-id at the `imp` level. See the following `imp` object.

Example:
```
{
            "id": "1",
            "bidfloor": 0.03,
            "banner": {
                "h": 250,
                "w": 300,
                "pos": 0
            },
            "ext": {
                "paf": {
                    "transaction-id": "4640dc9f-385f-4e02-a0e5-abbf241af94d"
                }
            }
        }
```

See the example in this spec.
https://github.com/prebid/addressability-framework/blob/main/mvp-spec/dsp-implementation.md

Reply:
that is being set on line 32 https://github.com/openx/Prebid.js/pull/46/files#diff-82f56fea551670affb017e76a242b6899257dab99443ed3222ae5297161aa2e2R32